stats/opentelemetry/internal/testutils/testutils.go

```diff
@@ -59,7 +59,16 @@ func waitForServerCompletedRPCs(ctx context.Context, t *testing.T, reader metric
 		if !ok {
 			continue
 		}
-		metricdatatest.AssertEqual(t, wantMetric, val, metricdatatest.IgnoreTimestamp(), metricdatatest.IgnoreExemplars())
+		if _, ok := val.Data.(metricdata.Histogram[int64]); ok {
+			if len(wantMetric.Data.(metricdata.Histogram[int64]).DataPoints) > len(val.Data.(metricdata.Histogram[int64]).DataPoints) {
+				continue
+			}
+		}
+		if _, ok := val.Data.(metricdata.Histogram[float64]); ok {
+			if len(wantMetric.Data.(metricdata.Histogram[float64]).DataPoints) > len(val.Data.(metricdata.Histogram[float64]).DataPoints) {
+				continue
+			}
```

Comment:
nit: Can use a type switch instead of `if` statements:
```go
switch data := val.Data.(type) {
case metricdata.Histogram[int64]:
	if len(wantMetric.Data.(metricdata.Histogram[int64]).DataPoints) > len(data.DataPoints) {
		continue
	}
case metricdata.Histogram[float64]:
	if len(wantMetric.Data.(metricdata.Histogram[float64]).DataPoints) > len(data.DataPoints) {
		continue
	}
}
```